### DISCUSSION TITLE
multi GPU DP traning encounter error, how to solve?

### DISCUSSION THREAD

**Discussion by tonystarkiss** [deepmodeling/deepmd-kit#974] [Q&A]
(base) root@bf9e9850e99e:/data/dpmd/deepmd-kit/examples/dpmd_raw/train# horovodrun -np 8  dp train --mpi-log=workers input.json
2021-08-15 12:48:07.708004: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
/bin/bash: /data/deepmd-kit-2.0.0/lib/libtinfo.so.6: no version information available (required by /bin/bash)
[mpiexec@bf9e9850e99e] match_arg (utils/args/args.c:163): unrecognized argument allow-run-as-root
[mpiexec@bf9e9850e99e] HYDU_parse_array (utils/args/args.c:178): argument matching returned error
[mpiexec@bf9e9850e99e] parse_args (ui/mpich/utils.c:1642): error parsing input array
[mpiexec@bf9e9850e99e] HYD_uii_mpx_get_parameters (ui/mpich/utils.c:1694): unable to parse user arguments
[mpiexec@bf9e9850e99e] main (ui/mpich/mpiexec.c:148): error parsing parameters

**Comment by njzjz** [comment]
@shishaochen

**Reply by tonystarkiss** [reply]
I have solved the issue, thanks, just reinstall with deepmd-kit-2.0.0.b4.
but I find a bug: when I run the following command, every GPU will create 4 processors, there are 16 processors in total, the np should be 1, the total processors are 4.
horovodrun -np 4 dp train --mpi-log=workers input.json

**Reply by tonystarkiss** [reply]
exit the base env will be ok.
conda deactivate
